### ISSUE TITLE
Fix --report-needless-disables to accept a boolean value and not a string

### ISSUE THREAD

**Issue by remcohaszing** [stylelint/stylelint#3818]
> Clearly describe the bug

According to the documentation, The CLI accepts a argument `--report-needless-disables`. However, it is treated by the truthiness of its string value.

For example:

```sh
# Lints directory
$ stylelint .
# Reports needless disables
$ stylelint --report-needless-disables any-non-empty-string .
# Reports needless disables
$ stylelint . --report-needless-disables
# Paths are missing. report-needless-disables now has the string value '.'
$ stylelint --report-needless-disables .
```

> Which rule, if any, is the bug related to?

None

> What CSS is needed to reproduce the bug?

None

> What stylelint configuration is needed to reproduce the bug?

None

> Which version of stylelint are you using?

`9.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

For the option `--report-needless-disables` to be treated as a boolean CLI argument

> What actually happened (e.g. what warnings or errors did you get)?

`--report-needless-disables` accepts a string value. If it’s non empty, it is treated as truthy.

**Comment by jeddy3** [comment]
@remcohaszing Thanks for the report and for using the template. Good catch. It looks like we've forgotten to give the [the flag](https://github.com/stylelint/stylelint/blob/master/lib/cli.js#L337-L339) a `type: boolean` property/value pair.
